coveo-functools/coveo_functools/flex/types.py

```diff
@@ -4,6 +4,7 @@
 
 
 JSON_TYPES = (
+    bytes,  # technically not a json type, but maximizes compatibility

```

Comment:
Hmmm... I'm concerned by this... Should we really accept invalid JSON? This is a surprising behavior. It also limit what we do on our side with it (for instance, we couldn't use anything from `json` module).

Also, from what I understand, this change is not directly linked with the goal of your pull request :slightly_smiling_face: .

Reply:
You're right, it's not related to this PR! But this actually fixes a bug.

Here's an example test:

```
def test_bytes() -> None:
    @dataclass
    class Test:
        a: bytes

    assert deserialize({"a": b"test"}, hint=Test).a == b"test"
    assert deserialize({"a": "test"}, hint=Test).a == "test"
```

Without my change, it breaks on the 2nd assert.

- On the first assert, it sees the annotation `bytes` as a custom class, and then the isinstance() is a success, so it's returned.
- On the second assert, it sees the annotation `bytes` as a custom class, and then the value is in the wrong type, so it will launch `hint(value)` and crash:

```
>       return hint(value)
E       TypeError: string argument without an encoding
```

This is like calling `bytes("test")`; it crashes.

By adding it to the JSON_TYPES, flex just stops handling bytes altogether but won't prevent you from using them. Similarly, if you feed a `str` in a field annotated with `int`, flex will not convert the string to int because we assume the json conversion already took care of that before. You will in fact have a string in there, even though it's annotated with int. 

Let me know if you still think this change impacts you!

Reply:
update: Actually, the test works in both the old and the new version. There is no such thing as `hint(value)` in the previous version of flex.